File: Leetcode/45. Jump Game II.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n,l,r=len(nums),0,0
        jumps = 0
        while r<n-1:
            far=0
            for i in range(l,r+1):
                far=max(nums[i]+i,far)
            l=r+1
            jumps=jumps+1
            r=far
        return jumps
        

File: Leetcode/45. Jump Game II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n=nums.size(),l=0,r=0;
        int jumps=0;
        while(r<n-1){
            int far=INT_MIN;
            for(int i=l;i<r+1;i++){
                far=max(i+nums[i],far);
            }
            l=r+1;
            jumps++;
            r=far;
        }
        return jumps;
    }
};
